tutorials/workflow/csharp/challenges-tips/IdempotentActivity.cs

```diff
@@ -0,0 +1,18 @@
+﻿using Dapr.Client;
+using Dapr.Workflow;
+
+namespace WorkflowApp;
+
+internal sealed class IdempotentActivity : WorkflowActivity<string, string>
+{
+    public override async Task<InventoryResult> RunAsync(WorkflowActivityContext context, OrderItem orderItem)
+    {
+        // Beware of non-idempotent operations in an activity.
+        // Dapr Workflow guarantees at-least-once execution of activities, so activities might be executed more than once.
```

Comment:
While true, I'd supplement with some context in that if the workflow activity successfully returns a value, it will not be run again as its value will be used from the workflow context. This makes for a better argument then that workflows should do as little as possible so they improve their success rate and minimize being executed more than once (but should they be, yes, designing for idempotency would be ideal).

Reply:
Will clarify the comment